packages/familie-header/src/søk/Søkeresultater.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+
+import styled from 'styled-components';
+
+import AlertStripe from 'nav-frontend-alertstriper';
+import navFarger from 'nav-frontend-core';
+import { Normaltekst } from 'nav-frontend-typografi';
+
+import { adressebeskyttelsestyper, Ressurs, RessursStatus } from '@navikt/familie-typer';
+
+import { Søkeresultat } from '../typer';
+
+import { inputId } from '.';
+import { formaterPersonIdent } from './formatter';
+
+interface Props {
+    formaterResultat?: (søkResultat: Søkeresultat, erSøkeresultatValgt: boolean) => React.ReactNode;
+    søkeresultatOnClick: (søkResultat: Søkeresultat) => void;
+    søkeresultater: Ressurs<Søkeresultat[]>;
+    valgtSøkeresultat: number;
+    settValgtSøkeresultat: (søkeresultatIndex: number) => void;
+}
+
+const ResultatListe = styled.ul`
+    width: 20rem;
+    padding: 0;
+    margin: 0;
+`;
+
+const StyledAlertStripe = styled(AlertStripe)`
+    width: 20rem;
+`;
+
+const ResultatListeElement = styled.li<{ fokus: boolean }>`
+    list-style-type: none;
+    padding: 0.5rem;
+    outline: ${({ fokus }) => (fokus ? `3px solid ${navFarger.fokusFarge}` : '')};
+
+    &:hover {
+        background-color: ${navFarger.navLysGra};
+        cursor: pointer;
+    }
+`;
+
+const ResultatListeElementKnapp = styled.div`
+    display: flex;
+    flex-direction: row;
+
+    & svg {
+        text-align: center;
+        padding-right: 0.5rem;
+    }
+`;
+
+const SøkResultater: React.FC<Props> = ({
+    formaterResultat,
+    settValgtSøkeresultat,
+    søkeresultatOnClick,
+    søkeresultater,
+    valgtSøkeresultat,
+}) => {
+    switch (søkeresultater.status) {
+        case RessursStatus.SUKSESS:
+            return søkeresultater.data.length > 0 ? (
+                <ResultatListe aria-labelledby={inputId}>
+                    {søkeresultater.data.map((søkResultat: Søkeresultat, index: number) => {
+                        if (formaterResultat) {
+                            return formaterResultat(søkResultat, index === valgtSøkeresultat);
+                        } else {
+                            return (
```

Comment:
RessursStatus.SUKSESS er vanskligt å lese i tilleg når det er i en switch-case statement runt om kring

Reply:
Hmm, litt usikker på om jeg skjønte denne kommentaren. Mener du at switch casen er vanskelig å forstå?

Reply:
Jeg syns att det er for mye logikk i denne komponenten

Reply:
mm, ikke uenig. Kanskje flytte ut Suksess caset i egen komponent da?